src/viam_mlmodelservice_triton_impl.cpp

```diff
@@ -419,6 +470,9 @@ class Service : public vsdk::MLModelService, public vsdk::Stoppable, public vsdk
         }
         state->model_name = std::move(*model_name_string);
 
+        const std::string path_to_store_data = initialize_directory_(state->model_name);
+	symlink_mlmodel_(*state.get(), path_to_store_data);
```

Comment:
Why pull these out into separate functions if this is the only place they are called?

Reply:
Because keeping smaller functions with specific names makes it way easier to figure out what's going on in the code. The set of defined variables within a function doesn't get polluted with a bunch of stuff from previous tasks (so it's easier to hold the whole thing in your head), and it's clear where tasks start and stop. The extra bytes spent on a vtable and extra cycles spent adding a new frame to the stack are negligible compared to the amount of thought a human can save while working on the code.